### PR TITLE
Add extension source documentation and nits

### PR DESCRIPTION
**Pull request by reneeleung**
Fixes nits from #1321

### CODE REVIEW

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 1]
```diff
@@ -1116,8 +1130,9 @@ class KotlinGenerator private constructor(
       if (empty) return null
     }
     return PropertySpec.builder(fieldName, optionsType.typeName)
-            .initializer(initializer)
-            .build()
+        .jvmField()
+        .initializer("\n%L", initializer)
+        .build()
```
Comment (oldergod) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: wasn't it only when we would have to generate java interop code?

Comment (reneeleung) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: yes, we thought it should work in interop mode

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 2]
```diff
@@ -660,6 +661,9 @@ class KotlinGenerator private constructor(
             if (field.documentation.isNotBlank()) {
               addKdoc("%L\n", field.documentation)
             }
+            if (field.isExtension) {
+              addKdoc("Extension source: %L\n", field.location.withPathOnly());
```
Comment (Egorand) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: nit: trailing `;` is redundant

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 3]
```diff
@@ -1246,17 +1263,17 @@ class KotlinGenerator private constructor(
...
     return builder.primaryConstructor(primaryConstructor.build())
-            .build()
+        .build()
   }
 
   private fun generateEnumCompanion(message: EnumType): TypeSpec {
+    val nameAllocator = nameAllocator(message)
     val companionObjectBuilder = TypeSpec.companionObjectBuilder()
-    val options = optionsField(ENUM_OPTIONS, "ENUM_OPTIONS", message.options)
+    val options = optionsField(ENUM_OPTIONS, nameAllocator["ENUM_OPTIONS"], message.options)
```
Comment (Egorand) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Nice

File: wire-tests/src/jvmKotlinInteropTest/proto-kotlin/com/squareup/wire/protos/kotlin/simple/SimpleMessage.kt [hunk 4]
```diff
@@ -613,7 +613,9 @@ class SimpleMessage(
     BUZ(3, true);
 
     companion object {
-      val ENUM_OPTIONS: EnumOptions = EnumOptions(
+      @JvmField
```
Comment (Egorand) [inline wire-tests/src/jvmKotlinInteropTest/proto-kotlin/com/squareup/wire/protos/kotlin/simple/SimpleMessage.kt]: Perfect!

**Pull request merged**